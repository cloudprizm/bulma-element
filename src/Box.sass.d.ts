export const box: string;
export const spinAround: string;

export type Block = "box"
export type Others = "spinAround"
export type All = Block | Others
export type BEM = Record<All, boolean>
declare let module: Record<All, string>
export default module