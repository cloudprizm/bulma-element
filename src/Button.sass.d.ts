export const button: string;
export const isLoading: string;
export const spinAround: string;
export const isFocused: string;
export const isActive: string;
export const icon: string;
export const isSmall: string;
export const isMedium: string;
export const isLarge: string;
export const isHovered: string;
export const isText: string;
export const isWhite: string;
export const isInverted: string;
export const isOutlined: string;
export const isBlack: string;
export const isLight: string;
export const isDark: string;
export const isPrimary: string;
export const isLink: string;
export const isInfo: string;
export const isSuccess: string;
export const isWarning: string;
export const isDanger: string;
export const isFullwidth: string;
export const isStatic: string;
export const isRounded: string;
export const buttons: string;
export const hasAddons: string;
export const isSelected: string;
export const isExpanded: string;
export const isCentered: string;
export const isRight: string;

export type Block = "button" | "icon" | "buttons"
export type Modifiers = "isLoading" | "isFocused" | "isActive" | "isSmall" | "isMedium" | "isLarge" | "isHovered" | "isText" | "isWhite" | "isInverted" | "isOutlined" | "isBlack" | "isLight" | "isDark" | "isPrimary" | "isLink" | "isInfo" | "isSuccess" | "isWarning" | "isDanger" | "isFullwidth" | "isStatic" | "isRounded" | "hasAddons" | "isSelected" | "isExpanded" | "isCentered" | "isRight"
export type Others = "spinAround"
export type All = Block | Modifiers | Others
export type BEM = Record<All, boolean>
declare let module: Record<All, string>
export default module