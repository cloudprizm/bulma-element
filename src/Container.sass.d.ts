export const container: string;
export const isFluid: string;
export const isWidescreen: string;
export const isFullhd: string;
export const spinAround: string;

export type Block = "container"
export type Modifiers = "isFluid" | "isWidescreen" | "isFullhd"
export type Others = "spinAround"
export type All = Block | Modifiers | Others
export type BEM = Record<All, boolean>
declare let module: Record<All, string>
export default module