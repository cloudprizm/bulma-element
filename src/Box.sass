@import "@hungry/bulma-theme/bulma-utils.sass";
@import "bulma/sass/elements/box.sass";
