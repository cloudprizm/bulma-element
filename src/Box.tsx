import { styledWithVariants, div } from '@hungry/sassy-react-component'

import { withRibbon, WithRibbon } from './Ribbon'
import { combineCSSDefinitions } from './modifiers'

import CSS, { BEM } from './Box.sass'

const asBulmaVariant =
  styledWithVariants<WithRibbon<BEM>>(
    combineCSSDefinitions(withRibbon, CSS))

export const Box =
  asBulmaVariant('box')
    (div)