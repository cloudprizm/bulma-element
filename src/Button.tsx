import styled from 'styled-components'
import {
  styledWithVariants,
  toStyledGenericFromStringOrJSX,
  toStyledGenericFromStyledFunction
} from '@hungry/sassy-react-component'

import { WithModifiers, combineCSSWithModifiers } from './modifiers'

import CSS, { BEM } from './Button.sass'

const withEmbeddedVariants =
  styledWithVariants<WithModifiers<BEM>>(
    combineCSSWithModifiers(CSS))

export const makeButton = withEmbeddedVariants('button')

export const Button = makeButton(
  toStyledGenericFromStringOrJSX('button'))

Button.displayName = 'Button'

export const SubmitButton = makeButton(
  toStyledGenericFromStyledFunction(
    styled
      .button
      .attrs({ type: 'submit' })))

SubmitButton.displayName = 'SubmitButton'