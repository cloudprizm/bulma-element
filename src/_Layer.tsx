// import { zLayers } from '@hungry/bulma-theme'
// import { style } from 'styled-system'

// export const Layer = reduce((acc, val: string) => ({
//   ...acc,
//   [val]: style({
//     prop: 'zIndex',
//     transformValue: identity(val)
//   })
// }), {})