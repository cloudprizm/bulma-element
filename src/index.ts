export * from './Button'
export * from './Container'
export * from './Content'
export * from './Title'
export * from './Box'
export * from './Ribbon'
export * from './Image'
export * from './Footer'
export * from './ProgressBar'
export * from './Notification'
export * from './Icon'
export * from './Tag'
export * from './Form'
export * from './modifiers'

export * from './Tag.compound'
export * from './Form.compound'

export * from './zIndex'

// TODO export variant props from each file to make it extensible as it is in compound
// export * from './Layer'