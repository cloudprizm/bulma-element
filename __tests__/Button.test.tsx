import * as React from 'react'
import renderer from 'react-test-renderer'
import { Button, SubmitButton } from '../src/Button'

jest.mock('../src/modifiers', () => ({
  combineCSSDefinitions: () => ({}),
  combineCSSWithModifiers: () => ({
    isActive: 'isActive-css',
    isBlack: 'isBlack-css',
    button: 'asButton-css'
  })
}))

test('render basic button', () => {
  const component = (
    <Button p={10} fontSize={10} isActive isBlack disabled as="button">
      Dupa szatana!
    </Button>
  )
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render submit button', () => {
  const component = (
    <SubmitButton p={10} isActive>Dupa szatana!</SubmitButton>
  )

  const wrapper = renderer.create(component).toJSON()
  expect(wrapper).toMatchSnapshot()
})

// TODO investigate - why font size does not appear - most likely because fontSize is a function
// import { mount } from 'enzyme'
// mount(component)
// console.log('@@@', a)

// import styled from 'styled-components'
// import { render } from 'react-testing-library'
// import { fontSize } from 'styled-system'

//  expect(tree).toHaveStyleRule('fontSize', '10')
//  expect(tree).toHaveStyleRule('padding', '10')

// test.only('test basic styles', () => {
//   // ${p => fontSize(p)}
//   const Component = styled.button`
//     ${fontSize}
//     ${p => fontSize(p)}
//     fontSize: 10
//   `
//   const { container } = render(<Component fontSize={10} />)
//   console.log('@@@', container.firstChild)

//   // const tree = renderer.create(styled(<Cmp fontSize={10} />)``).toJSON()
//   // console.log('@@@ -->', tree)
//   // expect(tree).toHaveStyleRule('fontSize', '10')
// })
