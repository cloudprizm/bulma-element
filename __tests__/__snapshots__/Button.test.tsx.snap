// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render basic button 1`] = `
.c0 {
  padding: 10px;
  font-size: 10px;
}

<button
  className="asButton-css isActive-css isBlack-css c0"
  disabled={true}
  fontSize={10}
>
  Dupa szatana!
</button>
`;

exports[`render submit button 1`] = `
.c0 {
  padding: 10px;
}

<button
  className="asButton-css isActive-css c0"
  type="submit"
>
  Dupa szatana!
</button>
`;
